dataprofiler/profilers/categorical_column_profile.py

```diff
@@ -43,6 +43,14 @@ def __init__(self, name: str | None, options: CategoricalOptions = None) -> None
         self.__calculations: dict = {}
         self._filter_properties_w_options(self.__calculations, options)
         self._top_k_categories: int | None = None
+
+        # Conditions to stop categorical profiling
+        self.max_sample_size_to_check_stop_condition = None
+        self.stop_condition_unique_value_ratio = None
+        self._stop_condition_is_met = False
+
+        self._unique_ratio = 0.0
+        self._unique_count = 0.0
```

Comment:
I suggest we update naming based on thoughts below:
`self._stopped_at_unique_count`
`self._stopped_at_unique_ratio`
or of the like